### PR TITLE
add baritone status variable

### PR DESCRIPTION
**Pull request by LordMZTE**

### CODE REVIEW

**Comment by LordMZTE** [comment]
closes #297

File: src/main/kotlin/me/zeroeightsix/kami/BaritoneIntegration.kt [hunk 1]
```diff
@@ -23,6 +22,9 @@ object BaritoneIntegration {
         }
     }
 
+    val mainProcess
+        get() = BaritoneAPI.getProvider().primaryBaritone.pathingControlManager.mostRecentInControl().kotlin
+
```
Comment (5HT2) [inline src/main/kotlin/me/zeroeightsix/kami/BaritoneIntegration.kt]: I'd argue a better variable name is needed, such as `recentControlProcess`, as `mainProcess` could be mistaken for `primaryBaritone`

File: src/main/kotlin/me/zeroeightsix/kami/Utils.kt [hunk 2]
```diff
@@ -230,3 +229,15 @@ data class Colour(val a: Float, val r: Float, val g: Float, val b: Float) {
 }
 
 fun VertexConsumer.colour(colour: Colour) = this.color(colour.r, colour.g, colour.b, colour.a)
+
+fun<K, V> MutableMap<K, V>.put(p: Pair<K, V>) {
+    put(p.first, p.second)
+}
```
Comment (5HT2) [inline src/main/kotlin/me/zeroeightsix/kami/Utils.kt:235]: Why is this needed?

**Review by 5HT2 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/me/zeroeightsix/kami/Utils.kt (hunk 2 above)
Comment (LordMZTE) [reply src/main/kotlin/me/zeroeightsix/kami/Utils.kt:235]: it makes it cleaner to add new entries to a map. it is used [here](https://github.com/LordMZTE/KAMI/blob/feature/baritonevariable/src/main/kotlin/me/zeroeightsix/kami/gui/text/VarMap.kt#L155)

Comment (5HT2) [reply src/main/kotlin/me/zeroeightsix/kami/Utils.kt:235]: I didn't see the pair, my bad. I thought you were just doing a put of two values :p

File: src/main/kotlin/me/zeroeightsix/kami/BaritoneIntegration.kt [hunk 3]
```diff
@@ -13,8 +13,7 @@ import me.zeroeightsix.kami.setting.KamiConfig
 import me.zeroeightsix.kami.setting.settingInterface
 
 object BaritoneIntegration {
-
-    private val present by lazy {
+    val present by lazy {
         try {
```
Comment (zeroeightysix) [inline src/main/kotlin/me/zeroeightsix/kami/BaritoneIntegration.kt]: This was private for a reason. Look at the `invoke` overload on BaritoneIntegration, you're supposed to check its value like so:

```kotlin
BaritoneIntegration {
    // This only runs if `present` is true!
}
```

File: src/main/kotlin/me/zeroeightsix/kami/gui/text/VarMap.kt [hunk 4]
```diff
@@ -147,7 +150,22 @@ object VarMap {
         "gametime_day_ticks" numeric { (mc.world?.timeOfDay?.toDouble() ?: 0.0) % 24000.0 },
         "realtime_ampm" string { ampmFormat.format(Date()) },
         "realtime_24h" string { militaryFormat.format(Date()) }
-    )
+    ).apply {
+        if (BaritoneIntegration.present) {
```
Comment (zeroeightysix) [inline src/main/kotlin/me/zeroeightsix/kami/gui/text/VarMap.kt]: Look at my earlier comment. No need to access `present` like this.

**Review by zeroeightysix — CHANGES_REQUESTED** [review]

File: src/main/kotlin/me/zeroeightsix/kami/BaritoneIntegration.kt [hunk 5]
```diff
@@ -13,7 +13,6 @@ import me.zeroeightsix.kami.setting.KamiConfig
 import me.zeroeightsix.kami.setting.settingInterface
 
 object BaritoneIntegration {
-
```
Comment (zeroeightysix) [inline src/main/kotlin/me/zeroeightsix/kami/BaritoneIntegration.kt]: Please restore this line

File: src/main/kotlin/me/zeroeightsix/kami/gui/text/VarMap.kt [hunk 6]
```diff
@@ -147,7 +150,22 @@ object VarMap {
         "gametime_day_ticks" numeric { (mc.world?.timeOfDay?.toDouble() ?: 0.0) % 24000.0 },
         "realtime_ampm" string { ampmFormat.format(Date()) },
         "realtime_24h" string { militaryFormat.format(Date()) }
-    )
+    ).apply {
+        BaritoneIntegration {
+            put("baritone" string { BaritoneIntegration.recentControlProcess?.displayName() ?: "" })
```
Comment (zeroeightysix) [inline src/main/kotlin/me/zeroeightsix/kami/gui/text/VarMap.kt]: I would probably prefer something like `"None"` instead of `""` as a default value.

In, for example, KAMI's default HUD - this would make a lot more sense:
```
(white)None (grey)baritone
```
instead of just a "floating" `baritone`:
```
(white) (grey)baritone
       ^ space here!
```

File: src/main/kotlin/me/zeroeightsix/kami/gui/text/VarMap.kt [hunk 7]
```diff
@@ -147,7 +150,22 @@ object VarMap {
...
+            put("baritone_formatted" string {
+                BaritoneIntegration.recentControlProcess?.displayName()
+                    // remove brackets
+                    ?.replaceAll("[]{}".asIterable(), " ")
+                    // add space after comma
+                    ?.replace(",", ", ")
+                    // remove duplicate spaces
+                    ?.replace(Regex("\\s+"), " ")
+                    // remove spaces before commas
+                    ?.replace(" ,", ",") ?: ""
```
Comment (zeroeightysix) [inline src/main/kotlin/me/zeroeightsix/kami/gui/text/VarMap.kt]: Same thing with the default value.

**Review by zeroeightysix — CHANGES_REQUESTED** [review]
My last gripe with this is that the variables are only present if baritone integration is present. This makes sense, but if the user (temporarily) removes baritone from the classpath, their HUD won't recognise those variables anymore and revert them to the default `No variable selected`.

Ultimately, they would still be present internally, but not shown in the UI if baritone isn't present. This shouldn't be too hard to do if you added some kind of `visibility` property to `Variable`. If you feel like that's not worth your time, or a task too daunting, feel free to skip it or do it in another PR.

**Comment by LordMZTE** [comment]
i don't think that belongs in this PR as its not really related. this should probably be done seperately

File: src/main/kotlin/me/zeroeightsix/kami/gui/text/VarMap.kt (hunk 6 above)
Comment (LordMZTE) [reply src/main/kotlin/me/zeroeightsix/kami/gui/text/VarMap.kt]: sorry for not replying to this. i completely overlooked it. but this is implemented now 👍

**Comment by LordMZTE** [comment]
sorry, looks like i got some line ending shenanigans in my latest commit 😅 
i think it is for the better tho, as the file got CRLF like all other files if I'm not mistaking

**Comment by LordMZTE** [comment]
any updates on this?

**Review by 5HT2 — CHANGES_REQUESTED** [review]
Fix the line endings, it makes it impossible to review

**Comment by LordMZTE** [comment]
fixed :D

**Review by zeroeightysix — APPROVED** [review]
Thanks!

**Review by 5HT2 — APPROVED** [review]
